crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -458,7 +459,13 @@ impl BaseClient {
 /// If any encrypted events are found after that one, store them in the RoomInfo
 /// too so we can use them when we get the relevant keys.
 #[cfg(feature = "e2e-encryption")]
-fn cache_latest_events(room: &Room, room_info: &mut RoomInfo, events: &[SyncTimelineEvent]) {
+async fn cache_latest_events(
+    room: &Room,
+    room_info: &mut RoomInfo,
+    events: &[SyncTimelineEvent],
+    changes: Option<&StateChanges>,
+    store: Option<&Store>,
```

Comment:
Would it make sense to use a single option for both? They're either both set, or none is.